plane/src/drone/docker/commands.rs

```diff
@@ -146,6 +191,40 @@ pub fn get_container_config_from_executor_config(
         .map(|(k, v)| format!("{}={}", k, v))
         .collect();
 
+    let binds: Option<Vec<String>> = match (&mount_base, &exec_config.mount) {
+        (_, None) | (_, Some(Mount::Bool(false))) => None,
+        (Some(base), Some(mount_option)) => {
+            let mount_path = match mount_option {
+                Mount::Bool(true) => {
+                    if let Some(key) = key {
+                        base.join(&key.key.name)
+                    } else {
+                        return Err(anyhow::anyhow!(
+                            "Key is required for Bool(true) mount option"
+                        ));
+                    }
+                }
+                Mount::Path(path) => base.join(path),
+                _ => return Err(anyhow::anyhow!("Invalid mount option")),
+            };
+            let canonical_mount_path = normalize_path(&mount_path);
+            if !canonical_mount_path.starts_with(base) {
+                return Err(anyhow::anyhow!(
+                    "Mount path {}, which we canonicalized as {}, is not under the specified mount base {}.",
+                    mount_path.to_string_lossy(),
+                    canonical_mount_path.to_string_lossy(),
+                    base.to_string_lossy()
+                ));
+            }
+            Some(vec![format!(
+                "{}:{}",
+                canonical_mount_path.to_string_lossy(),
+                PLANE_DATA_DIR
+            )])
+        }
+        _ => None,
```

Comment:
Even if we don't return an error to the user, we should at least stick a `tracing::warn!` in here if `mount_base` is `None` but `exec_config.mount` is Some.